test/planning/test_prm.py

```diff
@@ -10,18 +10,30 @@
 np.random.seed(1234)
 
 
-def test_construct_roadmap():
+def construct_roadmap_test(options):
+    # Initialize models, and construct and return a planner with the provided options
     model, collision_model, _ = load_models()
-    options = PRMPlannerOptions(max_construction_nodes=100)
-
-    # Initial the planner and do the sampling
     planner = PRMPlanner(model, collision_model, options=options)
     planner.construct_roadmap()
 
-    # Planning should succeed and should have the correct start and end poses
+    # Nodes and edges means construction was all peachy.
     assert len(planner.graph.nodes) > 0
     assert len(planner.graph.edges) > 0
 
+    return planner
+
+
+def test_construct_roadmap():
+    # Initialize the roadmap with a small number of nodes
+    options = PRMPlannerOptions(max_construction_nodes=100)
+    construct_roadmap_test(options)
+
+
+def test_construct_roadmap_prm_star():
+    # Initialize the roadmap using PRM*
+    options = PRMPlannerOptions(prm_star=True, max_construction_nodes=100)
+    planner = construct_roadmap_test(options)
```

Comment:
I did a similar thing for the RRT* tests, where after generating the graph, I checked that the paths for the "star" variant were <= the length of the plain one.

Would that work in this context?

Reply:
This is a really, really good question that I don't know the answer to... I just don't know of a way to get the same guarantees that come with the rewiring in RRT*. There is not guarantee PRM* will produce shorter paths after a small number of iterations - only that it will converge to more optimal paths as the sample count increases. The proof is in the pudding in Appendix C of [https://people.eecs.berkeley.edu/~pabbeel/cs287-fa19/optreadings/rrtstar.pdf](rrtstar.pdf).

For instance, I can make this pass or fail with the 2-DOF example without some creative parameter setting and different start/goal nodes. 
```
def test_prm_star():
    
    start_pose = np.array([0.0, 0.0])
    goal_pose = np.array([1.0, 1.0])
    
    options = PRMPlannerOptions(prm_star=False)
    planner_original = construct_roadmap_test(options)
    path_original = planner_original.plan(start_pose, goal_pose)
    assert path_original is not None
    
    options = PRMPlannerOptions(prm_star=True)
    planner_star = construct_roadmap_test(options)
    path_star = planner_star.plan(start_pose, goal_pose)
    assert path_star is not None
    
    # Should* be faster
    assert get_path_length(path_star) <= get_path_length(path_original)
```

But my understanding at this point is very cursory... I might play around with it more/do some more reading. I think we could have a very contrived example that does it?